src/ape_test/provider.py

```diff
@@ -240,11 +249,18 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
                 self.web3.eth.call(txn_params)
             except (ValidationError, TransactionFailed) as err:
                 vm_err = self.get_virtual_machine_error(err, txn=receipt)
-                raise vm_err from err
-
-            # If we get here, for some reason the tx-replay did not produce
-            # a VM error.
-            receipt.raise_for_status()
+                if txn.fail_on_revert:
+                    raise vm_err from err
+                else:
+                    receipt.error = vm_err
+
+            if txn.fail_on_revert:
+                # If we get here, for some reason the tx-replay did not produce
+                # a VM error.
+                receipt.raise_for_status()
+
+        if receipt.error:
+            logger.error(receipt.error)
```

Comment:
Wonder if a nicer refactor might be to always set `.error` and then use that within `.raise_for_status()`

Reply:
that works for some cases, I can do that, but other cases we revert before getting a receipt

Reply:
we'd lose the `from err` part potentially

Reply:
i made it at least alway set `.error` so when using debuggers, the receipt is more useful